src/test/kotlin/com/terraformation/backend/i18n/TimeZonesTest.kt

```diff
@@ -0,0 +1,157 @@
+          objectMapper.readValue<List<String>>(inputStream).toSet()
+        }
+    val unsupportedZones =
+        timeZones
+            .getTimeZoneNames(Locale.ENGLISH)
+            .keys
+            .minus(TimeZones.UTC)
+            .map { it.id }
+            .toSet()
+            .minus(browserZones)
+
+    assertEquals(emptySet<String>(), unsupportedZones)
+  }
+
+  @Test
+  fun `includes city name if there are multiple cities in the same zone`() {
+    val names = timeZones.getTimeZoneNames(Locale.ENGLISH)
+
+    assertEquals("Pacific Time - Los Angeles", names[ZoneId.of("America/Los_Angeles")])
+    assertEquals("Pacific Time - Vancouver", names[ZoneId.of("America/Vancouver")])
+  }
+
+  @Test
+  fun `omits city name if zone name is unique without it`() {
+    assertEquals(
+        "Pitcairn Time", timeZones.getTimeZoneNames(Locale.ENGLISH)[ZoneId.of("Pacific/Pitcairn")])
+  }
+
+  @Test
+  fun `removes redundant zone IDs`() {
+    // This is an alias for Europe/London
+    val jerseyTime = ZoneId.of("Europe/Jersey")
+    assertFalse(
+        jerseyTime in timeZones.getTimeZoneNames(Locale.ENGLISH), "Should have removed $jerseyTime")
+  }
+
+  @Test
+  fun `differentiates between DST and non-DST time zones`() {
+    val names = timeZones.getTimeZoneNames(Locale.ENGLISH)
+
+    // Boise changes its clocks twice a year
+    assertEquals("Mountain Time - Boise", names[ZoneId.of("America/Boise")])
+    // Phoenix doesn't
+    assertEquals("Mountain Standard Time - Phoenix", names[ZoneId.of("America/Phoenix")])
+  }
+
+  companion object {
+    /**
+     * Locales whose time zone renderings should be tested. This must always include all our
+     * supported languages, but may also include other languages to improve our confidence that the
+     * time zone rendering code is sufficiently general.
+     */
+    @JvmStatic
+    fun locales(): Stream<Locale> =
+        Stream.of(
+                "de", // German
+                "en", // US English
+                "en_GB", // British English - nearly but not completely identical to US English
+                "es", // Spanish
+                "fr", // French
+                "it", // Italian
+                "ja", // Japanese
+                "ko", // Korean
+                "pt_BR", // Brazilian Portuguese
+                "pt_PT", // Portuguese Portuguese
+                "ru", // Russian
+                "th", // Thai
+                "uk", // Ukrainian
+                "vi", // Vietnamese
+                "zh_CN", // Simplified Chinese
+                "zh_TW", // Traditional Chinese
+            )
+            .map { StringUtils.parseLocaleString(it) }
+
+    @JvmStatic fun nonEnglishLocales(): Stream<Locale> = locales().filter { it.language != "en" }
+
+    @JvmStatic
+    fun browsers(): Stream<String> =
+        Stream.of(
+            "Chrome 108.0.5359.124 (OS X 12.6)",
+            "Edge 108.0.1462.54 (Windows 11)",
+            "Safari 17614.1.25.9.10 (OS X 12.6)",
```

Comment:
guessing we periodically update this to catch unsupported time zones?

Reply:
Yeah, that's the theory. We'll see how well we keep up with this in practice, but I think it won't be a huge problem to fall behind (we'd run into trouble if a browser removed zones from its list, but not if it updates existing zones or adds new ones).